src/functions/index.js

```diff
@@ -0,0 +1,27 @@
+import { SUITS, VALUES } from '../constants';
+
+export function createDeck() {
+	let deck = []; 
+	for(let suitIndex = 0; suitIndex < SUITS.length; suitIndex++)	{
+		for(let valueIndex = 0; valueIndex < VALUES.length; valueIndex++) {
+			const card = { CardValue: VALUES[valueIndex], CardSuit: SUITS[suitIndex] };
+			deck = [...deck, card];
+		}
+	}
+	return deck;
+};
+
+export function shuffle(array) {
+  let currentIndex = array.length;
+  let tempArray = array;
```

Comment:
Aqui ainda continua com o mesmo problema de alterar o `array` passado como parametro. Você está fazendo `tempArray = array`, isso fala para o JS que o ponto na memoria dos dois é o mesmo, então tudo que fizer com um, vai acontecer com o outro.

Então para corrigir:
```suggestion
  const tempArray = [...array];
```